ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java

```diff
@@ -66,6 +69,8 @@ public class ResourceManager implements Closeable, IngestionResourceManager {
     private IngestionResource<QueueWithSas> queues;
     private IngestionResource<QueueWithSas> successfulIngestionsQueues;
     private IngestionResource<QueueWithSas> failedIngestionsQueues;
+    private TimerTask refreshIngestionAuthTokenTask;
+    private TimerTask refreshIngestionResourcesTask;
 
     ResourceManager(Client client, long defaultRefreshTime, long refreshTimeOnFailure, @Nullable CloseableHttpClient httpClient) {
```

Comment:
We're putting a `CloseableHttpClient` in a wrapper that isn't closeable (`new HttpClientWrapper(httpClient)`)!
Now we can't close it!

Reply:
This is going to change soon 
also this is a client from the user - he should close it himself

Reply:
That's true for the `ManagedStreamingIngestClient` flow.
But this is also called in `com.microsoft.azure.kusto.ingest.QueuedIngestClientImpl#QueuedIngestClientImpl(com.microsoft.azure.kusto.data.auth.ConnectionStringBuilder, com.microsoft.azure.kusto.data.HttpClientProperties, boolean)`
specifically via this code:
`this(csb, properties == null ? null : HttpClientFactory.create(properties), autoCorrectEndpoint)`

So in this flow, the `CloseableHttpClient httpClient` is **never** created by the user.